File: neetcode/python/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left_ptr, right_ptr = 0, len(numbers) - 1
        
        while left_ptr < right_ptr:
            if numbers[left_ptr] + numbers[right_ptr] > target:
                right_ptr -= 1
            elif numbers[left_ptr] + numbers[right_ptr] < target:
                left_ptr += 1
            else:
                return [left_ptr + 1, right_ptr + 1]
# @lc code=end

File: neetcode/go/167.two-sum-ii-input-array-is-sorted.go
/*
 * @lc app=leetcode id=167 lang=golang
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

// @lc code=start
func twoSum(numbers []int, target int) []int {
	left_ptr, right_ptr := 0, len(numbers)-1

	for left_ptr < right_ptr {
		if numbers[left_ptr]+numbers[right_ptr] == target {
			return []int{left_ptr + 1, right_ptr + 1}
		} else if numbers[left_ptr]+numbers[right_ptr] > target {
			right_ptr--
		} else {
			left_ptr++
		}
	}
	return nil
}

// @lc code=end

